error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2' and 'XXH3_accumulate_sse2'

When building e9455b2c204df00f31200392f96ac12197de8bf0 with GCC 14.1 under MSYS2/MinGW64 using `-Og -msse -msse2`, I get the following errors.

```
$ gcc -Og -msse -msse2 -c xxhash.c -o xxhash.o
In file included from xxhash.c:42:
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_64b_internal' at xxhash.h:5915:5,
    inlined from 'XXH3_hashLong_64b_default' at xxhash.h:5951:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_64b_internal' at xxhash.h:5915:5,
    inlined from 'XXH3_hashLong_64b_withSecret' at xxhash.h:5937:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_64b_internal' at xxhash.h:5915:5,
    inlined from 'XXH3_hashLong_64b_withSeed_internal' at xxhash.h:5974:16,
    inlined from 'XXH3_hashLong_64b_withSeed' at xxhash.h:5993:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_64b_internal' at xxhash.h:5915:5,
    inlined from 'XXH3_hashLong_64b_withSeed_internal' at xxhash.h:5980:16,
    inlined from 'XXH3_hashLong_64b_withSeed' at xxhash.h:5993:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_128b_internal' at xxhash.h:6732:5,
    inlined from 'XXH3_hashLong_128b_default' at xxhash.h:6758:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_128b_internal' at xxhash.h:6732:5,
    inlined from 'XXH3_hashLong_128b_withSecret' at xxhash.h:6775:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_128b_internal' at xxhash.h:6732:5,
    inlined from 'XXH3_hashLong_128b_withSeed_internal' at xxhash.h:6787:16,
    inlined from 'XXH3_hashLong_128b_withSeed' at xxhash.h:6805:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function 'XXH3_hashLong_internal_loop',
    inlined from 'XXH3_hashLong_128b_internal' at xxhash.h:6732:5,
    inlined from 'XXH3_hashLong_128b_withSeed_internal' at xxhash.h:6792:16,
    inlined from 'XXH3_hashLong_128b_withSeed' at xxhash.h:6805:12:
xxhash.h:5130:1: error: inlining failed in call to 'always_inline' 'XXH3_scrambleAcc_sse2': function not considered for inlining
 5130 | XXH3_scrambleAcc_sse2(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
      | ^~~~~~~~~~~~~~~~~~~~~
xxhash.h:5853:9: note: called from here
 5853 |         f_scramble(acc, secret + secretSize - XXH_STRIPE_LEN);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5852:9: note: called from here
 5852 |         f_acc(acc, input + n*block_len, secret, nbStripesPerBlock);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:4816:1: error: inlining failed in call to 'always_inline' 'XXH3_accumulate_sse2': function not considered for inlining
 4816 | XXH3_accumulate_##name(xxh_u64* XXH_RESTRICT acc,           \
      | ^~~~~~~~~~~~~~~~
xxhash.h:5127:34: note: in expansion of macro 'XXH3_ACCUMULATE_TEMPLATE'
 5127 | XXH_FORCE_INLINE XXH_TARGET_SSE2 XXH3_ACCUMULATE_TEMPLATE(sse2)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
xxhash.h:5860:9: note: called from here
 5860 |         f_acc(acc, input + nb_blocks*block_len, secret, nbStripes);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
```
$ gcc --version
gcc.exe (Rev3, Built by MSYS2 project) 14.1.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -624,7 +624,7 @@ jobs:
       - name: mingw64
         if: ${{ matrix.msystem == 'mingw64' }}
         run: |
-          PATH=/mingw64/bin:$PATH /mingw32/bin/mingw32-make clean test MOREFLAGS=-Werror
+          PATH=/mingw64/bin:$PATH /mingw32/bin/mingw32-make clean test gcc-og-test MOREFLAGS=-Werror
           PATH=/mingw64/bin:$PATH /mingw32/bin/mingw32-make -C tests/bench
           # Abort if result of "file ./xxhsum.exe" doesn't contain 'x86-64'.
           # Expected output is "./xxhsum.exe: PE32+ executable (console) x86-64, for MS Windows"

--- a/Makefile
+++ b/Makefile
@@ -361,7 +361,7 @@ clangtest: clean
 .PHONY: gcc-og-test
 gcc-og-test: clean
 	@echo ---- test gcc -Og compilation ----
-	CFLAGS="-Og -Wall -Wextra -Wundef -Wshadow -Wcast-align -Werror -fPIC" MOREFLAGS="-Werror" $(MAKE) all
+	CFLAGS="-Og -Wall -Wextra -Wundef -Wshadow -Wcast-align -Werror -fPIC" CPPFLAGS="-DXXH_NO_INLINE_HINTS" MOREFLAGS="-Werror" $(MAKE) all
 
 .PHONY: cxxtest
 cxxtest: clean

--- a/README.md
+++ b/README.md
@@ -157,7 +157,7 @@ The following macros can be set at compilation time to modify `libxxhash`'s beha
                          Defining this macro to 1 will mark all internal functions as `static`, allowing the compiler to decide whether to inline a function or not.
                          This is very useful when optimizing for smallest binary size,
                          and is automatically defined when compiling with `-O0`, `-Os`, `-Oz`, or `-fno-inline` on GCC and Clang.
-                         This might also increase performance depending on compiler and architecture.
+                         It may also be required to successfully compile using `-Og`, depending on compiler version.
 - `XXH_SIZE_OPT`: `0`: default, optimize for speed
                   `1`: default for `-Os` and `-Oz`: disables some speed hacks for size optimization
                   `2`: makes code as small as possible, performance may cry

